cherrypy/process/servers.py

```diff
@@ -121,6 +121,14 @@ def index(self):
 import sys
 import time
 import warnings
+import contextlib
+
+import portend
+
+
+class Timeouts:
```

Comment:
I remember that talk; I asked the [second question](https://youtu.be/o9pEzgHorH0?t=21m19s). I was there and I certainly absorbed it and factor it into my philosophy. It's nice to review it again. I'm a big fan of functools and functional programming. My first formal training was in Scheme. I do apply KISS frequently. However, I don't think Diederich's criticisms apply here. The purpose is clear without having to trace to the origin.

I do abhore `ASPECT_ATTRIBUTE=val` constants in code. They are crude and clumsy. For example, if you want to inspect all attributes relating to `ASPECT`, you have to query all globals in the module and find the ones whose names match a pattern. With `class Timeout`, you can find all timeout constants with `dir(Timeout)`. And yet, the amount of typing to reference the constant is identical (`Timeouts.free` vs. 'TIMEOUTS_FREE`). Actually, it's less because `_` requires a chord (on most keyboards) versus `.` which is available with a single stroke.

After your feedback, I was tempted to roll back to the previous declarations, but after considering this more in depth, I now feel strongly that this code should be incorporated, not only for the benefits that it provides, but also to challenge the reader's assumptions and demonstrate a valuable paradigm.